docs/recommendations.md

```diff
@@ -1,18 +1,139 @@
+
 ## Binding to Network Interfaces
-Always bind to localhost rather than to 0.0.0.0 or any interface, unless there is a specific need to do otherwise. Binding to localhost reduces the attack surface of your application by making it only accessible to devices on the same machine.
-Binding to 0.0.0.0 or "all" interfaces can make your application respond on all current and future network interfaces.
+
+Always bind to localhost rather than to 0.0.0.0 or any interface, unless there is a specific need to do otherwise.
+Binding to localhost reduces the attack surface of your application by making it only accessible to devices on the same
+machine. Binding to 0.0.0.0 or "all" interfaces can make your application respond on all current and future network
+interfaces.
 
 Issue: ([#19](https://github.com/open-telemetry/sig-security/issues/19#issue-1926445623))
+
+## Code signing / Provenance
+
+We recommend using [`sigstore`](https://www.sigstore.dev/) to create signatures for your SIG's deliverables, such as
+binaries, packages, and containers. In most cases, you can achieve that with GitHub's [build provenance attestation
+action](https://github.com/actions/attest-build-provenance).
+
+Example of actions for signing packages and binaries
+
+```yaml
+name: My release - binaries
+
+jobs:
+  release:
+    permissions:
+      id-token: write
+      attestations: write
+      packages: write
+      contents: read
+
+    steps:
+      - name: Attest Build Provenance
+        uses: actions/attest-build-provenance@897ed5eab6ed058a474202017ada7f40bfa52940 # v1.0.0
+        with:
+          subject-path: "my-artifact.tar.gz"
+```
+
+And the following for a container image:
+
+```yaml
+name: My release - container images
+
+jobs:
+  release:
+    permissions:
+      id-token: write
+      attestations: write
+      packages: write
+      contents: read
+    steps:
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@0d4c9c5ea7693da7b068278f7b52bda2a190a446 # v3.2.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Prepare container metadata
+        id: meta
+        uses: docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81 # v5.5.1
+        with:
+          images: |
+            ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+            type=sha
+
+      - name: Build and push the images
+        id: push
+        uses: docker/build-push-action@a254f8ca60a858f3136a2f1f23a60969f2c402dd # v6.4.0
+        with:
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Attest container images
```

Comment:
Does this work with docker hub attestations or only for github repositories?

E.g. in weaver - we use this: https://github.com/open-telemetry/weaver/blob/main/.github/workflows/publish-docker.yml#L51